platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultGradleEnterprisePluginAdapter.java

```diff
@@ -95,14 +103,28 @@ public void buildFinished(@Nullable Throwable buildFailure) {
                 public Throwable getFailure() {
                     return buildFailure;
                 }
+
+                @Override
+                public Collection<Problem> getProblems() {
+                    return DefaultGradleEnterprisePluginAdapter.this.getProblems(buildFailure);
+                }
             });
         }
     }
 
+    private @Nonnull Collection<Problem> getProblems(@Nullable Throwable buildFailure) {
+        ImmutableList.Builder<Problem> builder = ImmutableList.builder();
+        while(buildFailure != null) {
+            Collection<Problem> problems = problemsMapping.get(buildFailure);
+            builder.addAll(problems);
+            buildFailure = buildFailure.getCause();
```

Comment:
❌ I don't think this works for a `MultiCauseException`, where you'd have multiple causes.